### PR TITLE
Lot Number Revamp 📱

### PR DESCRIPTION
**Pull request by ashfurrow**
This is a fix for https://github.com/artsy/auctions/issues/333 and it does a few things:

- Adds support for using `lot_label` instead of `lot_number`, which allows us to use non-numerical lot numbers like "13A".
- Removes use of the `lotPosition` field as a proxy for lot number, which was never accurate anyway.

This is a WIP because I want to test on device in a variety of contexts, but here are some screenshots from the simulator:

simulator screen shot mar 28 2017 11 31 31 am
simulator screen shot mar 28 2017 11 31 28 am
simulator screen shot mar 28 2017 11 29 33 am
simulator screen shot mar 28 2017 11 29 31 am

### CODE REVIEW

**Comment by ashfurrow** [comment]
Forgot the unit tests 😅 We should probably wait for 3.2.0 to get submitted before merging anyhow.

**Comment by ashfurrow** [comment]
Oh, and to explain the changed reference images: we were previously not stubbing a number at all, which was interpreted as zero, Since we _are_ now stubbing the `lot_label` result from metaphysics, the appropriate number ("1") shows up.

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">You need to set the Sticker extensions’s plist version to 3.2.0</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by ashfurrow** [comment]
The Danger failure is fixed in another PR: https://github.com/artsy/eigen/pull/2245

File: Artsy/Networking/ARRouter.m [hunk 1]
```diff
@@ -1112,6 +1112,7 @@ + (NSURLRequest *)liveSaleStaticDataRequest:(NSString *)saleID role:(NSString *)
       reserve_status\
       low_estimate_cents\
       high_estimate_cents\
+      lot_label\
       lot_number\
```
Comment (acjay) [inline Artsy/Networking/ARRouter.m]: Should we still query `lot_number`?

Comment (ashfurrow) [inline Artsy/Networking/ARRouter.m]: Good catch! I'll amend.

**Comment by ashfurrow** [comment]
@acjay I opted not to add tests for alphanumeric lot numbers specifically since the type system protects us there.

**Comment by ashfurrow** [comment]
CI is intermittently failing, as usual, and it's making it hard to merge #2244 and  #2197. I'm going to merge locally, run the unit tests on my machine, and push if successful.

**Pull request merged**